src/parsers/manifestjson.js

```diff
@@ -239,7 +239,9 @@ export default class ManifestJSONParser extends JSONParser {
     // Arrays can be extremely verbose, this tries to make them a little
     // more sane. Using a regex because there will likely be more as we
     // expand the schema.
-    const match = error.dataPath.match(/^\/(permissions)\/([\d+])/);
+    const match = error.dataPath.match(
+      /^\/(permissions|optional_permissions)\/([\d+])/
```

Comment:
@rpl this one seems more elegant. The previous one was sloppy from my end, I'm sorry.